glib/src/gstring.rs

```diff
@@ -511,6 +579,32 @@ mod tests {
         assert_eq!(s.as_str(), "foo");
     }
 
+    #[test]
+    fn test_from_glib_container() {
+        unsafe {
+            let a: GString = FromGlibContainer::from_glib_container_num(
+                ffi::g_strdup("hello_world".as_ptr() as *const i8),
+                5,
+            );
+            assert_eq!("hello", a.as_str());
+            let b: GString = FromGlibContainer::from_glib_none_num("hello_world".as_ptr(), 5);
+            assert_eq!("hello", b.as_str());
+
+            let c: GString =
+                FromGlibContainer::from_glib_none_num(std::ptr::null() as *const i8, 0);
+            assert_eq!("", c.as_str());
+
+            let d: GString = FromGlibContainer::from_glib_none_num("".as_ptr(), 0);
+            assert_eq!("", d.as_str());
+
+            let e: GString = FromGlibContainer::from_glib_container_num(
+                ffi::g_strdup(std::ptr::null() as *const i8),
```

Comment:
`g_strdup()` returns `NULL` if passed `NULL` or not? And the cast should be unneeded

Reply:
Yes, per the function documentation 
> Duplicates a string. If str is NULL it returns NULL. The returned string should be freed with g_free() when no longer needed.

The cast is needed because otherwise I keep getting 
```
 FromGlibContainer::from_glib_none_num(std::ptr::null(), 0);
     |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ cannot infer type for struct `gstring::GString`
```

Reply:
`std::ptr::null::<i8>()` might be nicer than a cast then :)

Reply:
Aha. Thanks! I wasn't aware I can do this :)